docs/book/src/developer/providers/migrations/v1.4-to-v1.5.md

```diff
@@ -38,6 +38,8 @@ maintainers of providers and consumers of our Go API.
 - Introduced function `CollectInfrastructureLogs` at the `ClusterLogCollector` interface in `test/framework/cluster_proxy.go` to allow collecting infrastructure related logs during tests.
 - A `GetTypedConfigOwner` function has been added to the `sigs.k8s.io./cluster-api/bootstrap/util` package. It is equivalent to `GetConfigOwner` except that it uses the cached typed client instead of the uncached unstructured client, so `GetTypedConfigOwner` is expected to be more performant.
 - `ClusterToObjectsMapper` in `sigs.k8s.io./cluster-api/util` has been deprecated, please use `ClusterToTypedObjectsMapper` instead.
+- Control Plane providers objects, such as the Kubeconfig and Certificate Authority Secret resources, must be labelled with the key-value pair `cluster.x-k8s.io/cluster-name=${CLUSTER_NAME}`.
```

Comment:
This should be only required for the kubeconfig.

We intentionally implemented a fallback for the CA secrets (the ones documented here: https://cluster-api.sigs.k8s.io/tasks/certs/using-custom-certificates.html)

But for better performance its recommended

Reply:
The getCertificateSecret func in certificates.go https://github.com/kubernetes-sigs/cluster-api/pull/8940/files#diff-abaeb85035e8215f53acee88947192b6701ad36cda330df23d02e38d18e711a1

Reply:
Can you please also add a quick note under "KubeConfig management": https://cluster-api.sigs.k8s.io/developer/architecture/controllers/control-plane.html#kubeconfig-management

Thank you!

Reply:
I suppose it's ready for review, thanks, Stefan!